source/js/front/form.js

```diff
@@ -1,91 +1,132 @@
-                const removeButton = repeater.querySelector('.btn-repeater-remove');
-                addButton.addEventListener('click', (e) => {
-                    const subFieldsClone = repeater.querySelector('.sub-fields').cloneNode(true);
-                    subFieldsClone.querySelectorAll('input').forEach(field => {field.value = '';});
-                    repeater.insertBefore(subFieldsClone, addButton);
-                    if (repeater.querySelectorAll('.sub-fields').length > 1) {
-                        removeButton.classList.remove('u-display--none');
-                    }
-                });
-                removeButton.addEventListener('click', (e) => {
-                    const subFields =  repeater.querySelectorAll('.sub-fields');
-                    if (subFields.length > 1) {
-                        subFields[subFields.length - 1].remove();
-                    }
-                    if (subFields.length === 1) {
-                        removeButton.classList.add('u-display--none');
-                    }
+            return false;
+        });
+
+        if (conditionResult.every((x) => x === true)) {
+            field.classList.remove('u-display--none');
+            field.querySelectorAll('input').forEach((x) => {
+                x.required = true;
+            });
+        } else {
+            field.classList.add('u-display--none');
+            field.querySelectorAll('input').forEach((x) => {
+                x.required = false;
+            });
+        }
+    },
+    setupRepeaters: (form) => {
+        const repeaters = form.querySelectorAll('.js-repeater');
+        repeaters.forEach((repeater) => {
+            const addButton = repeater.querySelector('.btn-repeater-add');
+            const removeButton = repeater.querySelector('.btn-repeater-remove');
+            addButton.addEventListener('click', (e) => {
+                const subFieldsClone = repeater
+                    .querySelector('.sub-fields')
+                    .cloneNode(true);
+                subFieldsClone.querySelectorAll('input').forEach((field) => {
+                    field.value = '';
                 });
+                repeater.insertBefore(subFieldsClone, addButton);
+                if (repeater.querySelectorAll('.sub-fields').length > 1) {
+                    removeButton.classList.remove('u-display--none');
+                }
             });
+            removeButton.addEventListener('click', (e) => {
+                const subFields = repeater.querySelectorAll('.sub-fields');
+                if (subFields.length > 1) {
+                    subFields[subFields.length - 1].remove();
+                }
+                if (subFields.length === 1) {
+                    removeButton.classList.add('u-display--none');
+                }
+            });
+        });
+    },
+    setupRemoteSelect: (form) => {
+        if (eventintegration?.apiurl === undefined) {
```

Comment:
hmm .. i'm not 100% sure if undefined will work? (maybe it does) but will look better with if(!eventintegration?.apiurl) imo. 

On second thoughts this function depends on an outer scope variable (eventintegration), perhaps its better to include it as an argument and also destruct it (eg. setupRemoteSelect: (form, { apiurl }) => () then we can just make the check if (!apiurl) return.

Reply:
destructing the argument does seem like the better solution here tbh, but we should just be depending on soemthing inside the library and it's a bit confusing